src/collection-view.js

```diff
@@ -128,12 +128,36 @@ Marionette.CollectionView = Marionette.View.extend({
     return this;
   },
 
+  // Reorder DOM after sorting. When your element's rendering
+  // do not use their index, you can pass reorderOnSort: true
+  // to only reorder the DOM after a sort instead of rendering
+  // all the collectionView
+  reorder: function () {
+    var children = this.children;
+    var idx = children._indexByModel;
+    var views = children._views;
+    // get the views in the same order as the models
+    var arr = this.collection.models.map(function (model) {
+      var viewid = idx[model.cid];
+      var view = views[viewid];
+      return view.$el;
+    });
+    // transform into a jQuery array
+    var allEls = $(arr).map(function () { return this.toArray(); });
+    allEls.detach();
+    this.$el.append(allEls);
```

Comment:
Correct me if there is a gotcha here, but I'm pretty sure this can just be

```diff
-    // transform into a jQuery array
-    var allEls = $(arr).map(function () { return this.toArray(); });
-    allEls.detach();
-    this.$el.append(allEls);
+    this.$el.append(arr);
```